backend/chainlit/server.py

```diff
@@ -870,13 +871,67 @@ async def upload_file(
     assert file.filename, "No filename for uploaded file"
     assert file.content_type, "No content type for uploaded file"
 
+    validate_file_upload(file)
+
     file_response = await session.persist_file(
         name=file.filename, content=content, mime=file.content_type
     )
 
     return JSONResponse(content=file_response)
 
 
+def validate_file_upload(file: UploadFile):
```

Comment:
I would prefer to have validation functions raise a well-defined `Exception` and to have a docstring specifying it's behaviour.

For example, in [pydantic](https://docs.pydantic.dev/latest/concepts/validators/#field-validators) this is a `ValueError` and in [Django](https://docs.djangoproject.com/en/5.1/ref/validators/) it's a `ValidationError`. It should definitely not be a `HTTPException` as the file validation code should not have anything to do with HTTP, it's just validating an `UploadFile`.

We can catch validation errors and then re-raise them in a HTTP request handler (`upload_file`).